components/merkledb/src/indexes/proof_list/proof.rs

```diff
@@ -15,7 +15,7 @@
 pub use crate::ValidationError; // TODO Change for a type alias after EJB switching to rust > 1.36
 
 use exonum_crypto::Hash;
-use failure::Fail;
+use exonum_proto::failure::{self, Fail};
```

Comment:
Since `exonum_proto` dependency is optional, I don't think that importing `failure` from `exonum_proto` here is safe.